File: Programmers/17677.py
import copy

def solution(str1, str2):
    a = []
    b = []

    for i in range(len(str1) - 1):
        if str1[i].isalpha() and str1[i + 1].isalpha():
            a.append((str1[i] + str1[i + 1]).lower())

    for i in range(len(str2) - 1):
        if str2[i].isalpha() and str2[i + 1].isalpha():
            b.append((str2[i] + str2[i + 1]).lower())

    b_copy = copy.deepcopy(b)
    common  = 0
    for x in a:
        if x in b_copy:
            common += 1
            b_copy.remove(x)

    union = len(a) + len(b) - common
    
    if common == 0 and union == 0:
        answer = 1
    elif union == 0:
        answer = 0
    else:
        answer = common / union

    return int(answer * 65536)

File: Programmers/17677.js
function solution(str1, str2) {
  let arr1 = [];
  let arr2 = [];

  for (let i = 0; i < str1.length - 1; i++) {
    if (isAlpha(str1[i]) && isAlpha(str1[i + 1]))
      arr1.push(str1.slice(i, i + 2).toUpperCase());
  }
  for (let i = 0; i < str2.length - 1; i++) {
    if (isAlpha(str2[i]) && isAlpha(str2[i + 1]))
      arr2.push(str2.slice(i, i + 2).toUpperCase());
  }

  const len1 = arr1.length;
  const len2 = arr2.length;
  if (len1 === 0 && len2 === 0) return 65536;

  let common_count = 0;
  arr1.forEach((pair) => {
    const idx = arr2.indexOf(pair);
    if (idx >= 0) {
      common_count += 1;
      arr2.splice(idx, 1);
    }
  });

  return Math.floor((common_count / (len1 + len2 - common_count)) * 65536);
}

function isAlpha(c) {
  if ((c >= "A" && c <= "Z") || (c >= "a" && c <= "z")) return true;
  else return false;
}
